File: s3-inf-ppo/laboratoria/lab09/java/src/Zoo/Zoo.java
package Zoo;

import Zoo.Animals.Animal;

import java.util.ArrayList;

public final class Zoo {

    private ArrayList<Animal> animals = new ArrayList<>();
    private String name;

    public Zoo(String name) {
        this.name = name;
        Log.info(name + " created.");
    }

    public Zoo addAnimal(Animal animal) {
        animals.add(animal);
        Log.info(animal.getName() + " added to zoo.");
        return this;
    }

    public void feedAnimals(String food) {
        Log.info();
        Log.info("Feeding started.");

        for(Animal animal : animals) {
            try {
                animal.feed(food);
            } catch (Exception e) {
                Log.warning(e.getMessage());
            }
        }
    }

}

File: s3-inf-ppo/laboratoria/lab09/php/src/Zoo.php
<?php

namespace PPO\Zoo;

use Exception;
use PPO\Zoo\Animals\Animal;

final class Zoo {

	/** @var Animal[] $animals  */
	private $animals = [];
	private $name;

	public function __construct(string $name) {
		$this->name = $name;
		Log::info("$name created.");
	}

	public function addAnimal(Animal $animal): self {
		$this->animals[] = $animal;
		Log::info("{$animal->getName()} added to zoo.");
		return $this;
	}

	public function feedAnimals(string $food): void {
		Log::info();
		Log::info("Feeding started.");
		foreach($this->animals as $animal) {
			try {
				$animal->feed($food);
			} catch(Exception $e) {
				Log::warning($e->getMessage());
			}
		}
	}

}
